### PR TITLE
US193 Custom Look and Feel

### LINKED ISSUE #99: image in the browser tab should the THM
At the moment it seems to show with the Vue logo.

### LINKED ISSUE #106: Color Scheme / Theme
I like the layout already. However, the colors could use some tweaking. 

I'd like the top panel to be dark -- some grey-black (around #555555. The rest of the colors could be something like a Monokai pallete. I like the pink used in the range sliders. 

I think the current blue is a bit bleh, so I'd change that.

Also, not sure, but it *might* be nice to set the background of the sidebar in the form to another color than white?

Also, the top navbar, "The Halo Mod" should be "TheHaloMod".

### LINKED ISSUE #114: Add "Acknowledging" page
There's a page to this effect on the current website. This should be added as a button in top-right.

### LINKED ISSUE #123: Plot Formatting: make lines go above grid
At the moment, the grey grid lines go "over" the plot lines, which to me looks weird. I think the plot lines should be on top.

### LINKED ISSUE #125: log buttons
The log buttons have two deficiencies:

1. They should remember their state (so if you go to another plot choice, then return to the original, it should remember the state it was in).
2. They should have defaults for each plot choice. This information should already be in the KEYMAP (or whatever that is now). 

The first point is not necessary if it seems too clunky, but the second is important.

### PR DESCRIPTION
**Pull request by aneuhold**
## Overview
This PR implements US 193, custom look and feel. It implements every acceptance criteria except for the generic form fields re-formatting because that was done in #147. Most of this PR is visual, but there are some store changes to persist the currently selected axis, log/unlog, and theme. 

### Theming

Theming was implemented. This resolves #106. The exact colors for light and dark mode are still being hashed out with Steven, but as it stands you can see below the light and dark mode (with a toggle at the top with a tooltip which changes the icon based on which mode it is in). Light/dark mode persists through refreshes and applies to the rest of the website when it is changed. 

Light mode:

image

Dark mode:

image

Checkout #106 for more screenshots. 

Setting up themes requires that [SASS](https://sass-lang.com/) is able to be loaded in Vue as an option for the `<style>` sections, so that was added as a [pre-processor](https://vue-loader.vuejs.org/guide/pre-processors.html).

There was a section added to the client readme on how to change the look based on the currently selected theme mode as well, so that should be useful for anyone that needs to change the style of things when the theme changes. It was a bit hard to figure out at first so hopefully that is helpful. It isn't in the vue-material documentation so if there are any questions please feel free to ask, I had to peruse the source code of vue-material for quite some time to figure it out. 

### Favicon and icons and all that

Favicon was added and so was a manifest / icons for apps on phones etc. This resolves #99. See below for a screenshot of the browser tab:

image

### Site title

This was adjusted to "TheHaloMod" per request of Steven in #106.

### Plot rendering adjustments

The plot rendering was adjusted so that the bottom axis doesn't overflow and returns the correct height so that it won't overlap anymore. Also the legend shouldn't overlap outside of the box anymore either. In addition, the grid-lines go behind the plot lines now. Resolves #123.

Also fixed issue where the plot would try to render when there wasn't any data. 

### Small store changes

Now the idb-keyval contains 3 primary objects: `models`, `theme`, `plot`. This persists data across refreshes so the user doesn't need to select the same model again. Resolves #125 and addresses part of #95. The log persistence was a bit challenging so it could probably have a refactor sometime soon in the future, but it seems to work well for now. Primarily the issue is with the logic of when the plot re-renders after changing an axis selection. That got a bit confusing, so it was left as-is. That was not fixed in this PR so it is the same as it was before. 

### Model name is centered now

image

Addresses #95.

### Acknowledging page was added

Resolves #114.

## Includes:
- client/README.md 
- client/package-lock.json 
- client/package.json 
- client/public/android-chrome-192x192.png 
- client/public/android-chrome-512x512.png 
- client/public/apple-touch-icon.png 
- client/public/favicon-16x16.png 
- client/public/favicon-32x32.png 
- client/public/favicon.ico 
- client/public/index.html 
- client/public/site.webmanifest 
- client/src/App.vue 
- client/src/assets/logo.png 
- client/src/components/Graph/Plot.vue 
- client/src/components/Graph/index.vue 
- client/src/components/Models/Model.vue 
- client/src/components/Navbar.vue 
- client/src/constants/PLOT.js 
- client/src/constants/PLOT_AXIS_OPIONS.json → client/src/constants/PLOT_AXIS_OPTIONS.json 
- client/src/constants/themeOptions.js 
- client/src/main.js 
- client/src/router/index.js 
- client/src/utils/Store.js 
- client/src/utils/downloads.js 
- client/src/utils/latex.js 
- client/src/utils/plot.js 
- client/src/views/About.vue 
- client/src/views/Acknowledgement.vue 
- client/src/views/Home.vue 
- client/src/views/theme.scss 
- client/tests/unit/Graph.spec.js 
- client/tests/unit/Store.spec.js 
- client/webpack.config.js 
- client/yarn.lock 

## Developer Checklist:
- [x] The code follows the Quality Policy file on Google Drive
- [x] My code has been developer-tested and includes unit tests
- [x] I have considered proper use of exceptions
- [x] I have eliminated IDE warnings

## Notes:
- Webpack was set to 4.x because of a bug in Vue/vue-material that prevents anything from running when a theme is chosen. It seems to have no other impact on the site and works as intended. Even using the same webpack version for dev didn't do anything different.
- Extra un-used script section removed from `App.vue`.
- `PLOT_AXIS_OPIONS` was missing a `T` so that was fixed. 
- `MathJax.texReset()` was removed because it was causing errors and removing it didn't seem to change how the axis labels were rendered. 
- Link to the github repo was added in the about page

## Refs:
- [US 193](https://the-halo-mod.atlassian.net/jira/software/projects/HM/boards/1?selectedIssue=HM-193)

### CODE REVIEW

**Comment by steven-murray** [comment]
Thanks @aneuhold this is awesome! 

As for the colors, I reckon in light mode that we should still use the blue-green as the primary and pink as accent. The Dark Mode looks perfect. I love that you can switch!

**Comment by aneuhold** [comment]
@steven-murray updated! Glad that you are liking it so far. Feel free to leave feedback here 😄 .

File: client/src/components/Navbar.vue
Comment (Tengelma) [inline client/src/components/Navbar.vue:24]: It's not 100% clear that the github icon is associated with the Report Issue. Might need to make the icon smaller and make it part of the Report Issue label (to the right of it).

Comment (Tengelma) [reply client/src/components/Navbar.vue:24]: And since this is open source we may also just want a github icon that just sends you to the repo so... maybe no icon here?

**Review by Tengelma — CHANGES_REQUESTED** [review]
One little request related to the new navbar.

Also, not necessarily a change you need to make but I think light looks way better and more professional than dark and that including the toggle strips some of the identity away from the site's aesthetic. I know you're still working on the colors with Steven but figured I'd just throw that feedback out and see what everyone else feels.

File: client/src/components/Navbar.vue
Comment (aneuhold) [reply client/src/components/Navbar.vue:24]: Steven said that he loves the toggle. So unless he says to remove it, seems like the way to go is to do what he said on that one. The identity comment is a little more confusing. Could you elaborate on that a little further?

Potentially removing the icon could help yes. I like it like that and hovering over it indicates it is combined with the report issue label. See below:

image

 But with this one as well, Steven specifically requested it. Might be something to bring up at the meeting on Monday.

Comment (Tengelma) [reply client/src/components/Navbar.vue:24]: Haha I can appreciate that the identity thing is a little ambiguous. Basicallly, my thinking is that this is a utility with a very narrow focus and a general goal of fast comprehension / navigation / usage. It holds your hand and does everything for you. I think adding bells and whistles like personalized theming distracts from that goal, especially when the toggle for that feature is right next to things that are actually really important.

The question then becomes can we tuck it away somewhere where it doesn't share real estate with those things? But then you're designing more UI elements just for a light/dark mode toggle that no prospective user would really miss and you're going even further down the rabbit hole in the process.

Sooooooo... ya. Obviously just my opinion and if Steven wants it Steven wants it but that's my hot take. Like I said, not expecting you to change it just wanted to discuss it.

Comment (Tengelma) [reply client/src/components/Navbar.vue:24]: And yea, we can talk about the icon during the meeting. I think github icons generally take you to github repos and its a little antipattern to use it for something else but w/e.

I will upgrade my review to a comment without explicit approval lol

**Review by Tengelma — COMMENTED** [review]
See active conversation for feedback.

**Review by Tengelma — APPROVED** [review]
I guess it leaves it as request changes if you go neutral. Will approve I guess :man_shrugging:

**Review by aschnapp — CHANGES_REQUESTED** [review]
I'm getting a bug where my existing models disappear, but still show up in the plot. 
Screen Shot 2021-04-04 at 3 40 17 PM

**Comment by aschnapp** [comment]
> I'm getting a bug where my existing models disappear, but still show up in the plot.
> Screen Shot 2021-04-04 at 3 40 17 PM

Is this a problem with switching branches only, which means all I have to do is to reset everything, or this shouldn't happen at all?

**Comment by aschnapp** [comment]
Also you might want to change the keyval store to idb, and use a separate store for everything else that are not models. 
It'll require a refactor to the Store.js component to create multiple stores.

**Comment by aschnapp** [comment]
After looking into it a little more, I'll approve it.

**Review by aschnapp — APPROVED** [review]

**Pull request merged**
